### PR TITLE
API for Restoring backup for cloud volume

### PR DESCRIPTION
**Pull request by jeffibm**
Creating an API end point for restoring back up for cloud volume

API initiated from https://github.com/ManageIQ/manageiq-ui-classic/pull/8187

@miq-bot add-reviewer @Fryguy
@miq-bot add-label enhancement
@miq-bot assign @Fryguy

### CODE REVIEW

**Comment by kbrock** [comment]
kicked (ruby issue)

File: app/controllers/api/cloud_volumes_controller.rb [hunk 1]
```diff
@@ -47,5 +47,17 @@ def create_backup_resource(type, id, data)
         {:task_id => cloud_volume.backup_create_queue(User.current_userid, data)}
       end
     end
+
+    def restore_backup_resource(type, id, data)
+      api_resource(type, id, "Restoring backup", :supports => :backup_restore) do |cloud_volume|
+        {:task_id => cloud_volume.backup_restore_queue(User.current_userid, data)}
```
Comment (kbrock) [inline app/controllers/api/cloud_volumes_controller.rb:53]:
```suggestion
        {:task_id => cloud_volume.backup_restore_queue(User.current_userid, data.symbolize_keys)}
```

I think we need to do this.

File: app/controllers/api/cloud_volumes_controller.rb [hunk 2]
```diff
@@ -47,5 +47,17 @@ def create_backup_resource(type, id, data)
         {:task_id => cloud_volume.backup_create_queue(User.current_userid, data)}
       end
     end
+
+    def restore_backup_resource(type, id, data)
+      api_resource(type, id, "Restoring backup", :supports => :backup_restore) do |cloud_volume|
```
Comment (kbrock) [inline app/controllers/api/cloud_volumes_controller.rb:52]:
```suggestion
      api_resource(type, id, "Restoring backup for", :supports => :backup_restore) do |cloud_volume|
```

you read it "#{phrase} #{cloud_volume}". So when there are 2 words, we need to add the for... (not perfect, but here we are)

ditto below

**Review by kbrock — COMMENTED** [review]
lookin' good.
just minor nits.

I did re-kick the tests. strange errors over there.

**Comment by kbrock** [comment]
LGTM. I will merge once we remove the create_snapshot.
The snapshots have been moved to a subcollection

**Comment by jeffibm** [comment]
post /api/cloud_volumes/23
{
    "action": "restore_backup",
     "resource": {
         "backup_id": "1"
     }
}

{
    "results": [
        {
            "success": true,
            "message": "Restoring backup for Cloud Volume id: 23 name: 'ocp-master01-docker''",
            "task_id": "436071",
            "task_href": "http://localhost:3000/api/tasks/436071",
            "href": "http://localhost:3000/api/cloud_volumes/23"
        }
    ]
}

**Pull request merged**